src/CMakeLists.txt

```diff
@@ -285,9 +285,11 @@ set( EXIV2_SRC  exiv2.cpp
                 utils.hpp
 )
 
-add_executable( exiv2 ${EXIV2_SRC} )
-target_link_libraries( exiv2 PRIVATE exiv2lib )
-install(TARGETS exiv2 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+if(EXIV2_ENABLE_TOOLS)
```

Comment:
You have the correct idea in reorganizing the code.  We really should have src/ for the library code (test/src=test applications) and samples/src for sample applications.  However, this is lots of messing about.  Your thoughts involving  `if (EXIV2_ENABLE_TOOLS)` is the quick fix.

I'm got a couple of other things to suggest:
1) It should be called EXIV2_ENABLE_BUILD_EXIV2APP "Build exiv2 command line program"
2) The help string for EXIV2_ENABLE_BUILD_SAMPLES "Build sample applications"